include/zephyr/kernel.h

```diff
@@ -6153,6 +6153,186 @@ void k_sys_runtime_stats_enable(void);
+
+/**
+ * @brief Block signals in the accompanying @ref k_sig_set
+ *
+ * @see @ref k_sig_mask
+ */
+#define K_SIG_BLOCK 0
+
+/**
+ * @brief Set signals in the accompanying @ref k_sig_set
+ *
+ * @see @ref k_sig_mask
+ */
+#define K_SIG_SETMASK 1
+
+/**
+ * @brief Unblock signals in the accompanying @ref k_sig_set
+ *
+ * @see @ref k_sig_mask
+ */
+#define K_SIG_UNBLOCK 2
+
+/** @brief Minimum realtime signal number */
+#define K_SIG_RTMIN 32
+
+/** @brief Number of supported realtime signals */
+#define K_SIG_NUM_RT MAX(CONFIG_SIGNAL_SET_SIZE - K_SIG_RTMIN, 0)
+
+/**
+ * @brief Manipulate the mask of the calling thread.
+ *
+ * @param how how to manipulate the set
+ * @param set the set to manipulate
+ * @param[out] oset the resulting set
+ * @retval 0 on success
+ * @retval -EINVAL for invalid values of @a how
+ */
+__syscall int k_sig_mask(int how, const struct k_sig_set *set, struct k_sig_set *oset);
+
+/**
+ * @brief Get set of pending signals for the calling thread.
+ *
+ * @param[out] set the set of pending signals for the calling thread.
+ * @retval 0 on success
+ * @retval -EINVAL for invalid values of @a how
+ */
+__syscall int k_sig_pending(struct k_sig_set *set);
+
+/**
+ * @brief Queue a signal for process ID @a pid.
+ *
+ * Queue the signal @a signo for process ID @a pid with value @a value.
+ *
+ * This function is safe to call in ISR context.
+ *
+ * @param pid the process ID for which to queue the signal.
+ * @param signo the signal number to queue.
+ * @param value the value associated with the signal.
+ *
+ * @retval 0 on success.
+ * @retval -EAGAIN if there were insufficient resources within the specified @a timeout.
+ * @retval -EINVAL if @a signo is invalid.
+ * @return -EPERM if the calling thread lacks permissions to signal the process with id @a pid.
+ * @return -ESRCH if a process with id @a pid does not exist.
+ */
+__syscall int k_sig_queue(k_pid_t pid, int signo, union k_sig_val value);
+
+/**
+ * @brief Wait for any signal in @a set for a maximum duration specified by @a timeout.
+ *
+ * Wait for any signal in @a set to be delivered to the calling thread.
+ *
+ * This call does not block if @ref K_NO_WAIT is specified in the @a timeout parameter.
+ * This call blocks indefinitely if @ref K_FOREVER is specified in the @a timeout parameter.
+ *
+ * @param set set of signals to wait for.
+ * @param[out] info pointer to a location in which to store signal-related information.
+ * @param timeout upper bound time to wait for a signal in @a set to be delivered.
+ *
+ * @retval 0 on success.
+ * @retval -EAGAIN if no signal in @a set was received within the specified @a timeout.
+ * @retval -EINTR if the operation was interrupted by a signal.
+ * @retval -ENOSYS if dynamic thread stack allocation is disabled
+ * @retval -EWOULDBLOCK if the operation would block and was called in ISR context.
+ *
+ * @see CONFIG_DYNAMIC_THREAD
+ */
+__syscall int k_sig_timedwait(const struct k_sig_set *set, struct k_sig_info *info,
```

Comment:
`k_sigtimedwait()`?

Reply:
Since Jukka also thought `k_sig_` was maybe preferable, I've left it that way (can revisit if needed).